### PR TITLE
feat: minAmtPerSec per nftType 

### PR DESCRIPTION
**Pull request by manuelpolzhofer**
- different minAmtPerSec for each NFT type
- reduced limit and mint to uint64 for storage optimization (allows 2^64 potential NFTs per type)

### CODE REVIEW

File: src/nft.sol [hunk 1]
```diff
@@ -10,7 +10,8 @@ import {DaiPool} from "../lib/radicle-streaming/src/DaiPool.sol";
 
 struct InputNFTType {
     uint128 nftTypeId;
```
Comment (manuelpolzhofer) [inline src/nft.sol:12]: We could also change `uint128 nftTypeId;` to `uint64 nftTypeId;` to optimize the input struct array but input is cheaper than storage.

File: src/nft.sol [hunk 2]
```diff
@@ -20,25 +21,22 @@ contract FundingNFT is ERC721, Ownable {
...
     IERC20 public dai;
 
-    // minimum streaming amount per second to mint an NFT
-    uint128 minAmtPerSec;
-
     struct NFTType {
-        uint128 limit;
-        uint128 minted;
+        uint64 limit;
+        uint64 minted;
```
Comment (CodeSandwich) [inline src/nft.sol:26]: Another approach could be this :laughing: 
```suggestion
        uint128 unminted;
```

Comment (manuelpolzhofer) [reply src/nft.sol:26]: Mhm, we would loss on-chain the information about initial `limit`. If we use subtraction and un-minted. That was my reasoning. For 3rd parties to build on top thats an important information.

Off-chain obviously no problems with events to reconstruct it.

Comment (CodeSandwich) [reply src/nft.sol:26]: Ah, no, we're using `limit` non-zeroness to detect if an NFT type already exists.

Comment (manuelpolzhofer) [reply src/nft.sol:26]: I think 2^64 are enough max NFTs per type.

**Pull request merged**